### DISCUSSION TITLE
Passing a callback as a target

### DISCUSSION THREAD

**Discussion by atomicpages** [gilbarbara/react-joyride#860] [Ideas]
Rather than have state-driven `steps`, it'd be cool to have the `target` work with a callback. For ex:

```ts
import type { Props as JoyrideProps } from "react-joyride";

const steps: JoyrideProps["props"] = [
  target: () => {
    return document.querySelector("...");
  },
  // ...
];
```

The expectation here is callback execution is synchronous. While not "the react way", it would go a long way in terms of usability. If this is something folks are interested in, I'd be happy to make a PR for this!

**Comment by gilbarbara** [comment]
Hey @atomicpages 

>  it would go a long way in terms of usability.

How so?
What is the advantage of this approach over the existing one? Since it's synchronous, I can't see the benefits.

**Reply by atomicpages** [reply]
The target could be calculated when the step mounts rather than storing all the steps in state and remounting the component
